From f704632d2ff440c5a8e13cc06a7fce13829a70dc Mon Sep 17 00:00:00 2001
Message-Id: <f704632d2ff440c5a8e13cc06a7fce13829a70dc.1428007496.git.feitong.yi@intel.com>
In-Reply-To: <96b4927b026b1cf56e9298f537d7ca517a5902b4.1428007496.git.feitong.yi@intel.com>
References: <96b4927b026b1cf56e9298f537d7ca517a5902b4.1428007496.git.feitong.yi@intel.com>
From: Namrta <namrta.salonie@intel.com>
Date: Fri, 20 Mar 2015 22:54:47 +0530
Subject: [PATCH 2/6] SQUASHME! [VPG]: drm/i915: Fix for object dereference
 after free.

Moved kfree to after usage of object by perfmon.
Found by static analysis tool.

SQUASHME! - This patch should be squashed into the following existing patch:
commit c5441a12ce23b3b9f9d5cb5f0e676f1fe5fdc085
Change-Id: I33fcc622edd5c5321916e037521aaac025f42339
Author: Adam Rutkowski <adam.j.rutkowski@intel.com>
Date:   Mon Sep 8 15:02:02 2014 -0400

    REVERTME [VPG]: drm/i915: Refactoring perfmon IRQ support

Issue: GMINL-7287
Change-Id: I5eb8da77e1d13f993c8c4a195723ac43417d3c78
Signed-off-by: Namrta <namrta.salonie@intel.com>
---
 drivers/gpu/drm/i915/i915_dma.c |    3 +--
 1 file changed, 1 insertion(+), 2 deletions(-)

diff --git a/drivers/gpu/drm/i915/i915_dma.c b/drivers/gpu/drm/i915/i915_dma.c
index 22d917e..c52cc38 100644
--- a/drivers/gpu/drm/i915/i915_dma.c
+++ b/drivers/gpu/drm/i915/i915_dma.c
@@ -2066,10 +2066,9 @@ int i915_driver_unload(struct drm_device *dev)
 	i915_cmd_parser_cleanup(dev_priv);
 
 	pci_dev_put(dev_priv->bridge_dev);
-	kfree(dev_priv);
 
 	i915_perfmon_cleanup(dev_priv);
-
+	kfree(dev_priv);
 	return 0;
 }
 
-- 
1.7.9.5

